File: May 2022/May_03.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int n = nums.size(), i = 0, j = n - 1;
        vector<int> sorted(n);
        for(int i = 0; i < n; i++)
            sorted[i] = nums[i];
        sort(sorted.begin(), sorted.end());
        while(i <= j && sorted[i] == nums[i])
            i++;
        while(i <= j && sorted[j] == nums[j])
            j--;
        return j - i + 1;
    }
};

File: May 2022/May_03.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        int[] sorted = nums.clone();
        Arrays.sort(sorted);
        int i = 0, j = nums.length - 1;
        while (i <= j && nums[i] == sorted[i])
            i++;
        while(j >= i && nums[j] == sorted[j])
            j--;
        return j - i + 1;
    }
}
